CRM/Member/BAO/Membership.php

```diff
@@ -2508,4 +2516,93 @@ public static function getContactsCancelledMembership($contactID, $isTest = FALS
     return $cancelledMembershipIds;
   }
 
+  /**
+   *
+   * Given specified contact, return a map of membeship orgs (that contact
+   * has ever been member of), and info on the last membership that contact
+   * has signed up for this org.
+
+   * Initial use case for this method was to return a list of membership
+   * type that a contact should be renewed on when renewing using *any*
+   * priceset, unlike online renewals which only renew for a specific price set
+   *
+   * @param int $contactId
+   * @return array
+   *   Array (int $member_of_contact_id =>
+   *      Array (of the fields returned by the results set (see below)
+   *   )
+   *
+   */
+  public static function getContactMemberhipsByMembeshipOrg($contactId) {
+    $query = "
+          select org.member_of_contact_id,
+            org.display_name,
+            mem.id as membership_id,
+            mem.membership_type_id,
+            mem.status_id,
+            st.is_current_member,
+            li.contribution_id,
+            li.price_field_id,
+            li.price_field_value_id,
+            co.receive_date
+          from (select distinct member_of_contact_id, con.display_name from civicrm_membership_type left join civicrm_contact con on con.id = member_of_contact_id) org
```

Comment:
Just a stylistic not - we normally use the mysql convention of uppercasing FROM, INNER JOIN, WHERE, ORDER BY